salt/utils/process.py

```diff
@@ -807,11 +807,15 @@ def _handle_signals(self, signum, sigframe):
         msg += '. Exiting'
         log.debug(msg)
         if HAS_PSUTIL:
-            process = psutil.Process(self.pid)
-            if hasattr(process, 'children'):
-                for child in process.children(recursive=True):
-                    if child.is_running():
-                        child.terminate()
+            if psutil.pid_exists(self.pid):
+                try:
+                    process = psutil.Process(self.pid)
+                    if hasattr(process, 'children'):
+                        for child in process.children(recursive=True):
+                            if child.is_running():
+                                child.terminate()
```

Comment:
Sorry for not mentioning this earlier, but maybe it's worth adding a try-catch here as well, if any children fail to terminate (due to race-conditions), to still try to terminate the rest of the child (not to stop on the first one that fails). Or maybe it's okay to stop on the first one?

Reply:
@rares-pop @dmurphy18 I don't think we need this `if psutil.pid_exists(self.pid):` at all. Just let it raise an exception if the parent doesn't exist. However, when iterating through the children I think we should except each one individually in case one goes away but others don't.

Maybe move this functionality to a utility method and add some unit tests for it.

Reply:
@dwoz - that was exactly my feedback too.  👍 

The first comment was about the unneeded pid_exists (I mentioned the possible race-condition), and the second about having the try-raise for children.